### ISSUE TITLE
CA1802 could also flag some instance fields

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#37982]
**Version Used**: 
3.3.0-beta3-19409-05+ed92d532473db83c1db47b313ee1c1bd7520aa08

**Steps to Reproduce**:
```C#
using System;

class Program
{
    static void Main()
    {
        foreach (string s in new Splitter().Split("hello,world"))
            Console.WriteLine(s);
    }
}

class Splitter
{
    private readonly char _separator = ',';

    public string[] Split(string s) => s.Split(_separator);
}
```

**Expected Behavior**:
CA1802 would suggest refactoring `_separator` to be `const`.

**Actual Behavior**:
CA1802 ignores `_separator` because it's an instance rather than static field.

**Comment by CyrusNajmabadi** [comment]
Agreed.  There woudl be a concern if these were public fields.  However, if they're private the scope of how they're used is known, and contained only to the assembly.  As such, making const is fine.  If there is public scope, I'd be hesitant making 'public' since then it could lead to a breaking change fro a user to switch from const to static-readonly in the future.

**Comment by stephentoub** [comment]
> There woudl be a concern if these were public fields.

Yes.  Though that applies to static fields as well.
